pkg/auth/iam/iam.go

```diff
@@ -137,6 +135,24 @@ func (filter *Filter) Auth(opts ...FilterOption) restful.FilterFunction {
 			}
 		}
 
+		if filter.options.SubdomainValidationEnabled {
+			if valid := validateSubdomainAgainstNamespace(getHost(req.Request), claims.Namespace, filter.options.SubdomainValidationExcludedNamespaces); !valid {
+				svcErr := respondError(http.StatusNotFound, SubdomainMismatch, "data not found: "+ErrorCodeMapping[SubdomainMismatch])
```

Comment:
why not simply use error handler like this:

```
  logIfErr(resp.WriteHeaderAndJson(http.StatusNotFound, ErrorResponse{
    ErrorCode:    SubdomainMismatch,
    ErrorMessage: "data not found: " + ErrorCodeMapping[SubdomainMismatch],
  }, restful.MIME_JSON))
```

From this current code, looks like it try to parse it twice: 
1. convert to `restful.ServiceError`
2. convert again to `ErrorResponse`

wdyt?